Web/test_HttpRequest.py

```diff
@@ -11,37 +16,82 @@ def recv(self, bufsize, flags=0):
 
 
 def test_http_request_host():
     oneTagsHeader = fakeSocket(b'GET /bank/Main-page HTTP/1.1\r\nHost: www.ryanbanco.com.br\r\n\r\n')
-    request = httpRequest.httpRequest(oneTagsHeader)
-    assert len(request.head().keys()) == 1
+    request = httpRequest.getNextHttpRequest(oneTagsHeader, logging)
+    assert len(request.getHeader()) == 1
 
 
 def test_http_request_calling_host():
     oneTagsHeader = fakeSocket(b'GET /bank/Main-page HTTP/1.1\r\nHost: www.ryanbanco.com.br\r\n\r\n')
-    request = httpRequest.httpRequest(oneTagsHeader)
-    assert request.head()[b'Host'] == b'www.ryanbanco.com.br'
+    request = httpRequest.getNextHttpRequest(oneTagsHeader, logging)
+    assert request.getHeader()[b'Host'] == b'www.ryanbanco.com.br'
 
 def test_http_request_calling_2_headers():
     oneTagsHeader = fakeSocket(b'GET /bank/Main-page HTTP/1.1\r\nHost: www.ryanbanco.com.br\r\nUser-Agent: Mozilla 5.0\r\n\r\n')
-    request = httpRequest.httpRequest(oneTagsHeader)
-    assert request.head()[b'User-Agent'] == b'Mozilla 5.0'
+    request = httpRequest.getNextHttpRequest(oneTagsHeader, logging)
+    assert request.getHeader()[b'User-Agent'] == b'Mozilla 5.0'
 
 def test_http_request_calling_3_headers():
     oneTagsHeader = fakeSocket(b'GET /bank/Main-page HTTP/1.1\r\nHost: www.ryanbanco.com.br\r\nUser-Agent: Mozilla 5.0\r\nAccept-Language: en-US\r\n\r\n')
-    request = httpRequest.httpRequest(oneTagsHeader)
-    assert request.head()[b'Accept-Language'] == b'en-US'
+    request = httpRequest.getNextHttpRequest(oneTagsHeader, logging)
+    assert request.getHeader()[b'Accept-Language'] == b'en-US'
+
+def test_http_request_getFirstLine_methodGet():
+    socket = fakeSocket(b'GET /bank/Main-page HTTP/1.1\r\n')
+    method, _, _ = httpRequest.getFirstLine(socket, logging)
+    assert method == "GET"
+
+def test_http_request_getFirstLine_methodPost():
+    socket = fakeSocket(b'POST /bank/Main-page HTTP/1.1\r\n')
+    method, _, _ = httpRequest.getFirstLine(socket, logging)
+    assert method == "POST"
+
+def test_http_request_getFirstLine_resource():
+    socket = fakeSocket(b'GET /bank/Main-page HTTP/1.1\r\n')
+    _, resource, _ = httpRequest.getFirstLine(socket, logging)
+    assert resource == "/bank/Main-page"
+
+def test_http_request_getFirstLine_version():
+    socket = fakeSocket(b'GET /bank/Main-page HTTP/1.1\r\n')
+    _, _, version = httpRequest.getFirstLine(socket, logging)
+    assert version == "1.1"
+
+def test_http_request_getBody():
+    oneTagHeader = {b'Content-length': b'20'}
+    socket = fakeSocket(b'a'*20)
+    body = httpRequest.getBody(socket, oneTagHeader, logging)
+    assert body == b'a'*20
+
+def test_http_request_getBody2():
+    oneTagHeader = {b'Content-length': b'20'}
+    socket = fakeSocket(b'a'*50)
+    body = httpRequest.getBody(socket, oneTagHeader, logging)
+    assert body == b'a'*20
+
+def test_http_request_getBody3():
+    oneTagHeader = {b'Content-length': b'20'}
+    socket = fakeSocket(b'a'*15)
+    try:
```

Comment:
pytest know how to assert exceptions : https://docs.pytest.org/en/stable/reference/reference.html#pytest.raises
Use it.